tests/query_async.cc

```diff
@@ -0,0 +1,51 @@
+#include <boost/test/unit_test.hpp>
+
+#include "config.hh"
+#include "scmd.hh"
+#include "fixture.hh"
+
+#define ARGS(id) (id), 115, 3456, 104755, 123341233124, 30.75, 32342.23, std::string("sadasdasda")
+#define TYPES int64_t, int8_t, int16_t, int32_t, int64_t, float, double, std::string
+#define ARGS_TUPLE(id) std::make_tuple(ARGS(id))
+#define RESULT_TUPLE(r) std::make_tuple(r.get_column<int64_t>("key"), \
+                                        r.get_column<int8_t>("a"),    \
+                                        r.get_column<int16_t>("b"), \
+                                        r.get_column<int32_t>("c"), \
+                                        r.get_column<int64_t>("d"), \
+                                        r.get_column<float>("e"), \
+                                        r.get_column<double>("f"), \
+                                        r.get_column<std::string>("g"))
+
+
+
+BOOST_FIXTURE_TEST_SUITE(async_query, scylla_fixture)
+
+BOOST_AUTO_TEST_CASE(insert)
+{
+    scmd::statement stmt("INSERT INTO test_keyspace.test_table (key, a, b, c, d, e, f, g) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?);", 8);
+    stmt.bind<TYPES>(ARGS(2 << 16 | 1));
+    auto future = session->execute_async(stmt);
+    future.wait();
+}
+
+BOOST_AUTO_TEST_CASE(select)
+{
+    scmd::statement stmt("INSERT INTO test_keyspace.test_table (key, a, b, c, d, e, f, g) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?);", 8);
+    stmt.bind<TYPES>(ARGS(2 << 16 | 2));
```

Comment:
What is this? ARGS(2 << 16 | 2) is unreadable.

Reply:
Well, I wanted to have unique ids for rows. So I take some number for test case and put it in higher bits, and some number for row in test case in lower bits. Could you propose a nice way to accomplish this (unique row numbers)? I'm open to ideas.

Reply:
int id = generateId();
stmt.bind<TYPES>(id, ARGS);
or
stmt.bind<int, TYPES>(id, ARGS)